spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/Predictor.scala

```diff
@@ -78,4 +79,58 @@ class Predictor[T: ClassTag] private[optim](
       })
     }
   }
+
+
+  /**
+   * model predict images, return imageFrame with predicted tensor
+   * @param imageFrame imageFrame that contains images
+   * @param outputLayer if outputLayer is not null, the output of layer that matches
+   *                      outputLayer will be used as predicted output
+   * @param shareBuffer whether to share same memory for each batch predict results
+   * @param batchPerPartition batch size per partition, default is 4
+   * @param predictKey key to store predicted result
+   */
+  def predictImage(imageFrame: ImageFrame,
+    outputLayer: String = null,
+    shareBuffer: Boolean = false,
+    batchPerPartition: Int = 4,
+    predictKey: String = ImageFeature.predict): DistributedImageFrame = {
+    require(imageFrame.isDistributed(), "please provide a distributed imageframe")
+    // share convolution fInput
+    SpatialShareConvolution.shareConvolution(model)
+    val rdd = imageFrame.asInstanceOf[DistributedImageFrame].rdd
+    val modelBroad = ModelBroadcast[T]().broadcast(rdd.sparkContext, model.evaluate())
+    val partitionNum = rdd.partitions.length
+    val toBatchBroad = rdd.sparkContext.broadcast(SampleToMiniBatch(
+      batchSize = partitionNum * batchPerPartition,
+      partitionNum = Some(partitionNum)), shareBuffer)
+    val result = rdd.mapPartitions(partition => {
+      val localModel = modelBroad.value()
+      val localToBatch = toBatchBroad.value._1.cloneTransformer()
+
+      partition.grouped(batchPerPartition).flatMap(imageFeatures => {
+        val validImageFeatures = imageFeatures.filter(_.isValid)
+        val samples = validImageFeatures.map(x => x[Sample[T]](ImageFeature.sample))
+        val batch = localToBatch(samples.toIterator).next()
```

Comment:
Instead of using next, maybe it's safer to do another flatmap here? See https://github.com/intel-analytics/BigDL/blob/master/spark/dl/src/main/scala/org/apache/spark/ml/DLEstimator.scala#L301